skills/src/main/java/cm/aptoide/skills/games/BackgroundGameService.kt

```diff
@@ -0,0 +1,145 @@
+  companion object {
+    private const val NOTIFICATION_SERVICE_ID = 77798
+    private const val CHANNEL_ID = "game_notification_channel_id"
+    private const val CHANNEL_NAME = "Game Notification Channel"
+
+    private const val ACTION_DISMISS = "ACTION_DISMISS"
+    private const val SESSION = "SESSION"
+    private const val GET_ROOM_PERIOD_SECONDS = 3L
+    private const val PLAYER_SEPARATOR = " vs "
+
+    @JvmStatic
+    fun newIntent(context: Context, sessionToken: String) =
+        Intent(context, BackgroundGameService::class.java).apply {
+          putExtra(SESSION, sessionToken)
+        }
+  }
+
+  @Inject
+  lateinit var roomRepository: RoomRepository
+
+  private lateinit var notificationManager: NotificationManager
+  private lateinit var periodicGameChecker: PeriodicGameChecker
+
+  override fun onBind(intent: Intent): IBinder? {
+    return null
+  }
+
+  override fun onCreate() {
+    super.onCreate()
+    notificationManager = getNotificationManager()
+    periodicGameChecker = PeriodicGameChecker(roomRepository, GET_ROOM_PERIOD_SECONDS, this)
+  }
+
+  private fun getNotificationManager(): NotificationManager {
+    val notificationManager =
+        this.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      val importance = NotificationManager.IMPORTANCE_HIGH
+      val notificationChannel = NotificationChannel(CHANNEL_ID, CHANNEL_NAME, importance)
+      notificationManager.createNotificationChannel(notificationChannel)
+    }
+    return notificationManager
+  }
+
+  override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+    when (intent?.action) {
+      ACTION_DISMISS -> stopService()
+    }
+
+    val session: String? = intent?.getStringExtra(SESSION)
+    session?.let {
+      periodicGameChecker.start(it)
+      val notification = getNotification(
+          getString(R.string.playing_game_notification_title),
+          getString(R.string.playing_game_notification_body)
+      )
+      this.startForeground(NOTIFICATION_SERVICE_ID, notification)
+    }
+    return super.onStartCommand(intent, flags, startId)
+  }
+
+  fun getNotification(title: String, text: String): Notification {
+    val notificationBuilder = NotificationCompat.Builder(this, CHANNEL_ID)
+    val dismissIntent = createNotificationDismissIntent()
+    return notificationBuilder.setContentTitle(title)
+        .addAction(0, getString(R.string.dismiss_button), dismissIntent)
+        .setSmallIcon(R.drawable.ic_launcher_foreground)
+        .setDeleteIntent(dismissIntent)
+        .setContentText(text)
+        .build()
+  }
+
+  private fun createNotificationDismissIntent(): PendingIntent {
+    val intent = Intent(this, BackgroundGameService::class.java)
+    intent.action = ACTION_DISMISS
+    return PendingIntent.getService(this, 0, intent, 0)
+  }
+
+  override fun onUpdate(roomResponse: RoomResponse) {
```

Comment:
if you only need a string (provided by `getFormattedGameDetails` method) why are you passing the hole network model to view?

Reply:
you're right. changed.